src/HealthChecks.Publisher.ApplicationInsights/DependencyInjection/ApplicationInsightsHealthCheckBuilderExtensions.cs

```diff
@@ -15,12 +15,14 @@ public static class ApplicationInsightsHealthCheckBuilderExtensions
         /// indicating the health check status (1 - Healthy, 0 - Unhealthy) and the total time the health check took to execute in milliseconds.
         /// </remarks>
         /// <param name="builder">The <see cref="IHealthChecksBuilder"/>.</param>
-        /// <param name="instrumentationKey">Specified Application Insights instrumentation key. Optional. If <c>null</c> TelemetryConfiguration.Active is used.</param>
+        /// <param name="connectionString">Specified Application Insights connection string. Optional. If <c>null</c> <see cref="TelemetryConfiguration"/> is resolved from DI container.</param>

```

Comment:
>If <c>null</c> <see cref="TelemetryConfiguration"/> is resolved from DI container

Are you sure?

Reply:
Yes, look at the code here: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/afc0157238a300bc0540bb88aead5bed9ef9b30c/src/HealthChecks.Publisher.ApplicationInsights/ApplicationInsightsPublisher.cs#L105

And here: https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/afc0157238a300bc0540bb88aead5bed9ef9b30c/src/HealthChecks.Publisher.ApplicationInsights/DependencyInjection/ApplicationInsightsHealthCheckBuilderExtensions.cs#L31

Reply:
I mean `TelemetryConfiguration` is **always** resolved from DI container (first options object, then Value from it in HC ctor) even if `connectionString` was specified. It is just not used in that case.

Reply:
```csharp
var configuration = string.IsNullOrWhiteSpace(_connectionString)
    ? string.IsNullOrWhiteSpace(_instrumentationKey)
        ? _telemetryConfiguration
        : new TelemetryConfiguration(_instrumentationKey)
    : new TelemetryConfiguration { ConnectionString = _connectionString };
```

The injected `_telemetryConfiguration` is only used if `_connectionString` and `_instrumentationKey` are null, empty or whitespace.

Reply:
Yes. I think I said the same. I mean _resolved from DI container_ != _used_ and it's worth to rephrase xml comment for `connectionString` argument to not confuse the reader.

Reply:
Ah, gotcha. I updated the XML comment.